src/result_processor.c

```diff
@@ -392,8 +395,11 @@ static int rpsortNext_innerLoop(ResultProcessor *rp, SearchResult *r) {
   int rc = rp->upstream->Next(rp->upstream, self->pooledResult);
 
   // if our upstream has finished - just change the state to not accumulating, and yield
-  if (rc == RS_RESULT_EOF || (rc == RS_RESULT_TIMEDOUT && rp->parent->timeoutPolicy == TimeoutPolicy_Return)) {
-    // Transition state:
+  if (rc == RS_RESULT_TIMEDOUT && (rp->parent->timeoutPolicy == TimeoutPolicy_Return)) {
+    self->timedOut = true;
+    rp->Next = rpsortNext_Yield;
+    return rpsortNext_Yield(rp, r);
+  } else if (rc == RS_RESULT_EOF) {
```

Comment:
These are the same changes as in #4165 - needed them here as well, and with the pace PRs are getting merged from the queue.. I just put it here as well.